lib/azure/armrest/configuration.rb

```diff
@@ -0,0 +1,132 @@
+module Azure
+  module Armrest
+    class Configuration
+      # Used to store unique token information.
+      @@tokens = Hash.new([])
+
+      # The client ID used to gather token information.
+      attr_accessor :client_id
+
+      # The client key used to gather token information.
+      attr_accessor :client_key
+
+      # The tenant ID used to gather token information.
+      attr_accessor :tenant_id
+
+      # The subscription ID used in requests and to gather token information.
+      attr_accessor :subscription_id
+
+      # The resource group used for a given http request.
+      attr_accessor :resource_group
+
+      # The API version used for a given http request. The default is 2015-01-01.
+      attr_accessor :api_version
+
+      # The grant type. The default is client_credentials
+      attr_accessor :grant_type
+
+      # The content type specified for http requests. The default is 'application/json'
+      attr_accessor :content_type
+
+      # The accept type specified for http request results. The default is 'application/json'
+      attr_accessor :accept
+
+      # The bearer token passed to all http requests.
+      attr_writer :token
+
+      # Set the time in which the token expires.
+      attr_writer :token_expiration
```

Comment:
combine them into one method like `def set_token(token, expiration)` as they need to be set externally at the same time